hw05/num2.c

```diff
@@ -0,0 +1,159 @@
+#include <stdio.h>
+#include <malloc.h>
+#include <string.h>
+#include "stack.h"
+
+#define SIZE 101
+
+bool calculator(Element **head, const char operand, int *errorCode) {
+    int secondElement = peek(*head);
+    pop(head);
+    int firstElement = peek(*head);
+    pop(head);
+
+    switch (operand) {
+        case '-': {
+            int tempResult = firstElement - secondElement;
+            *head = push(*head, tempResult, errorCode);
+            if (*errorCode == -1) {
+                return false;
+            }
+            break;
+        }
+        case '+': {
+            int tempResult = firstElement + secondElement;
+            *head = push(*head, tempResult, errorCode);
+            if (*errorCode == -1) {
+                return false;
+            }
+            break;
+        }
+        case '*': {
+            int tempResult = firstElement * secondElement;
+            *head = push(*head, tempResult, errorCode);
+            if (*errorCode == -1) {
+                return false;
+            }
+            break;
+        }
+        case '/': {
+            if (secondElement == 0) {
+                *errorCode = -1;
+                return false;
+            }
+            int tempResult = firstElement / secondElement;
+            *head = push(*head, tempResult, errorCode);
+            if (*errorCode == -1) {
+                return false;
+            }
+            break;
+        }
+        default:
+            break;
+    }
+    return true;
+}
+
+int processingExpression(const char expression[], int *errorCode) {
+    int size = (int) strlen(expression);
+    Element *stack = NULL;
+    for (int i = 0; i < size; ++i) {
+        switch (expression[i]) {
+            case '+':
+            case '-':
+            case '*':
+            case '/': {
+                bool correct = calculator(&stack, expression[i], errorCode);
+                if (!correct) {
+                    return -1;
+                }
+                break;
+            }
+            default:
+                if (expression[i] != ' ') {
+                    stack = push(stack, expression[i] - '0', errorCode);
+                    if (*errorCode == -1) {
+                        return -1;
+                    }
+                }
+                break;
+        }
+    }
+    if (sizeStack(stack) > 1 || isEmpty(stack)) {
+        *errorCode = -1;
+    }
+
+    int result = peek(stack);
+    removeStack(&stack);
+    return result;
+}
+
+bool testsCalculator() {
```

Comment:
```suggestion
bool testsCalculator(void) {
```
И тоже надо глаголом в повелительной форме